sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1693,8 +1704,18 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
                 // access columns from the source table.
                 val resolvedInsertCondition = insertCondition.map(
                   resolveExpressionByPlanOutput(_, m.sourceTable))
-                val assignments = targetTable.output.map { attr =>
-                  Assignment(attr, UnresolvedAttribute(Seq(attr.name)))
+                val assignments = if (m.schemaEvolutionEnabled) {
```

Comment:
@cloud-fan i was a bit back and forth on whether to do this by default.  It makes more cases pass that use to fail.  
I was thinking some data source already depend on the old behavior: https://docs.delta.io/latest/delta-update.html#automatic-schema-evolution 

Behavior without schema evolution is to throw the error.

Let me know , though if i should do it in all cases

Reply:
This is a good point. I think it's clearer to always do it, but this means v2 sources need to handle more cases now, so the error message may get worse if the behavior is undefined for those existing v2 sources.

Reply:
oh actually it doesn't matter. V2 sources that support custom schema evolution will likely use `ACCET_ANY_SCHEMA` capability, and this whole MERGE clauses resolution code path will be skipped for `ACCET_ANY_SCHEMA`.

So the question is: does the row level operation framework support missing columns in MERGE INSERT/UPDATE clauses?

Reply:
sounds good, let's do it.  Updated